tests/Module/CheckoutTest.php

```diff
@@ -0,0 +1,578 @@
+											[
+												'weight' => 20.0,
+												'price'  => 234.34,
+											],
+										],
+								],
+						],
+					'productDisallowedRateIds'           => [],
+					'productCategoryDisallowedRateIds'   => [],
+					'allowedCarrierNames'                =>
+						[
+							25061 => 'CZ Zásilkovna do auta',
+						],
+					'isCarDeliveryEnabled'               => true,
+					'isAgeVerificationRequiredByProduct' => true,
+					'cartWeight'                         => 1.0,
+				],
+			'allowed carrier names argument must support null'        =>
+				[
+					'expectedRateCount'                  => 1,
+					'carriers'                           =>
+						[
+							DummyFactory::createCarDeliveryCarrier(),
+						],
+					'carriersOptions'                    =>
+						[
+							'packetery_carrier_25061' =>
+								[
+									'id'                  => '25061',
+									'name'                => 'CZ Zásilkovna do auta',
+									'active'              => true,
+									'free_shipping_limit' => null,
+									'weight_limits'       =>
+										[
+											[
+												'weight' => 20.0,
+												'price'  => 234.34,
+											],
+										],
+								],
+						],
+					'productDisallowedRateIds'           => [],
+					'productCategoryDisallowedRateIds'   => [],
+					'allowedCarrierNames'                => null,
+					'isCarDeliveryEnabled'               => true,
+					'isAgeVerificationRequiredByProduct' => false,
+					'cartWeight'                         => 1.0,
+				],
+		];
+	}
+
+	/**
+	 * @dataProvider rateCreationDataProvider
+	 * @throws \PHPUnit\Framework\MockObject\Exception
+	 */
+	public function testGetShippingRates(
+		int $expectedRateCount,
+		array $carriers,
+		array $carriersOptions,
+		array $productDisallowedRateIds,
+		array $productCategoryDisallowedRateIds,
+		?array $allowedCarrierNames,
+		bool $isCarDeliveryEnabled,
+		bool $isAgeVerificationRequiredByProduct,
+		float $cartWeight
+	): void {
+		$wcProduct = $this->createMock( \WC_Product::class );
+		$wcProduct
+			->method( 'get_price' )
+			->willReturn( '100.00' );
+		$wcProduct
+			->method( 'get_category_ids' )
+			->willReturn( [ 1 ] );
+
+		$cartItem = [
+			'product_id' => 1,
+			'quantity'   => 1,
+			'data'       => $wcProduct,
+		];
+
+		$product = $this->createMock( Product\Entity::class );
+		$product
+			->method( 'isPhysical' )
+			->willReturn( true );
+		$product
+			->method( 'isAgeVerification18PlusRequired' )
+			->willReturn( $isAgeVerificationRequiredByProduct );
+		$product
+			->method( 'getDisallowedShippingRateIds' )
+			->willReturn( array_merge( [ Carrier\OptionPrefixer::getOptionId( 'anyDisallowedOnProduct' ) ], $productDisallowedRateIds ) );
+
+		$cart = $this->createMock( \WC_Cart::class );
+		$cart
+			->method( 'get_coupons' )
+			->willReturn( [] );
+
+		$wpAdapter                    = $this->getPacketeryMockFactory()->createWpAdapter();
+		$wcAdapter                    = $this->createMock( WcAdapter::class );
```

Comment:
`$wcAdapter` - a) create variable on line 449, c) usage on line 459 and b) definition of lines 468-513?

I like to a) create variable, b) definition of mock and c) use the variable.